x-pack/plugins/security_solution/public/detections/components/osquery/osquery_flyout.tsx

```diff
@@ -22,11 +29,44 @@ export interface OsqueryFlyoutProps {
   onClose: () => void;
 }
 
-export const OsqueryFlyout: React.FC<OsqueryFlyoutProps> = ({ agentId, onClose }) => {
+const TimelineComponent = React.memo((props) => {
+  return <EuiButtonEmpty {...props} size="xs" />;
+});
+TimelineComponent.displayName = 'TimelineComponent';
+
+export const OsqueryFlyoutComponent: React.FC<OsqueryFlyoutProps> = ({ agentId, onClose }) => {
   const {
-    services: { osquery },
+    services: { osquery, timelines },
   } = useKibana();
 
+  const { getAddToTimelineButton } = timelines.getHoverActions();
+
+  const handleAddToTimeline = useCallback(
+    (payload: { query: [string, string]; isIcon?: true }) => {
+      const [field, value] = payload.query;
+      const providerA: DataProvider = {
+        and: [],
+        enabled: true,
+        excluded: false,
+        id: value,
+        kqlQuery: '',
+        name: value,
+        queryMatch: {
+          field,
+          value,
+          operator: ':',
+        },
+      };
+
+      return getAddToTimelineButton({
+        dataProvider: providerA,
+        field: value,
+        ownFocus: false,
+        ...(payload.isIcon ? { showTooltip: true } : { Component: TimelineComponent }),
```

Comment:
is this ever called with isIcon false?

Reply:
No, never. It's either `true` or undefined